File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/LongInputFieldView.java
package com.saizad.mvvm.components.form.ui;

import android.content.Context;
import android.util.AttributeSet;

import com.sa.easyandroidform.ObjectUtils;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class LongInputFieldView extends InputFieldView<Long> {

    public LongInputFieldView(@NotNull Context context) {
        super(context);
    }

    public LongInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public LongInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public Long resolveFrom(@NotNull CharSequence charSequence) {
        try {
            return Long.valueOf(charSequence.toString());
        }catch (NumberFormatException e){
            return null;
        }
    }

    @Override
    public CharSequence resolve(@Nullable Long value) {
        if(ObjectUtils.isNull(value)){
            return null;
        }
        return value.toString();
    }

    @Override
    public boolean isSame(Long value, Long prevValue) {
        return value.equals(prevValue);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/LongInputFieldView.kt
package com.saizad.mvvm.components.form.ui

import android.content.Context
import android.util.AttributeSet
import com.sa.easyandroidform.ObjectUtils

open class LongInputFieldView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : InputFieldView<Long>(context, attrs, defStyleAttr) {


    override fun resolveFrom(charSequence: CharSequence): Long? {
        return try {
            java.lang.Long.valueOf(charSequence.toString())
        } catch (e: NumberFormatException) {
            null
        }
    }

    override fun resolve(value: Long?): CharSequence? {
        return if (ObjectUtils.isNull(value)) {
            null
        } else value.toString()
    }

    override fun isSame(value: Long, prevValue: Long): Boolean {
        return value == prevValue
    }
}